AddedMatlabAnalysisFunctionsAN/Bootstrapping.m

```diff
@@ -0,0 +1,52 @@
+%A script to perform bootstrapping on the data in peaks
+%in order to get a better estimate of the error
+% function [aver, err] = BootStrapping(matfile) % returns a scalar, which is the value of the error across all the peaks in a given matfile, calculated using bootstrapping
+% Niter - number of times to pool from the data set
+% nsample - size of the sample to pool from the data set
+% load outall_1_new.mat;
+% useonly - if want to exclude some mages from the analysis; not really used
+% col - which column to normalize by the DAPI column in the 'peaks' cell array). The
+% numbering refers to the numbering of peaks{ii}(:,col) and needs to be
+% verified before plotting , i.e. which channel corresponds to which
+% exactly column in peaks cell array.
+function [aver, err, alldata]=Bootstrapping(peaks,Niter,nsample,col)
+
+
+%if ~exist('useonly','var')
+ %   useonly=1:length(peaks);
+%end
+
+nlines=zeros(length(peaks),1);
+for ii=1:length(peaks)
+    nlines(ii)=size(peaks{ii},1);
+end
+alllines = sum(nlines);
+alldata=zeros(alllines,1);
+q=1;
+size(peaks{2},2);
+for ii=1:length(peaks)
+    if ~isempty(peaks{ii}) && size(peaks{ii},2)>10 %if ismember(ii,useonly) &&  ~isempty(peaks{ii})
+        if length(col)==1
+            alldata(q:(q+nlines(ii)-1))=peaks{ii}(:,col);%make a single column vector from all the data (normalized intensity of col.6 in peaks to dapi (col. 5) in peaks
+        else
+            alldata(q:(q+nlines(ii)-1))=peaks{ii}(:,col(1))./peaks{ii}(:,col(2));
+        end
+            q=q+nlines(ii);                                            
+    end
+    
+end
+dat =[];
+
+for j=1:Niter
+    for k=1:nsample
+        dat(k,1)=alldata(randi(length(alldata))); % populate the sample with randomly chosen elements(with resampling)of the 'initial' vector
+        
+    end
+    dataver(j)=mean(dat);
+    
+end
```

Comment:
the inner loop is unnecessary. if you are clever, you can probably get rid of the outer loop too.